sensors/sensors.go

```diff
@@ -0,0 +1,33 @@
+// Package sensors implements the high level structure of how sensors
+// are defined in slam libraries
+package sensors
+
+import (
+	"github.com/pkg/errors"
+
+	"go.viam.com/slam/config"
+)
+
+// Sensor contains the necessary information to define a generic sensor.
+type Sensor struct {
+	ID                string
+	Index             int
```

Comment:
What is the sensor index?

Reply:
Also, how will this work when sensors is a dictionary?

Reply:
The index refers to the position of the sensor in the `sensors` array. If we change `sensors` to be a dictionary, this will be replaced with an `ID`.